src/lib.rs

```diff
@@ -376,6 +387,13 @@ impl<T: Clone + Eq> Watcher for Direct<T> {
         self.state.value.clone()
     }
 
+    fn get_ref(&mut self) -> &Self::Value {
+        if let Some(shared) = self.shared.upgrade() {
+            self.state = shared.state();
```

Comment:
`get` has the same issue, but while we're at it, we should first check that `shared.state.read().unwrap().epoch` is actually higher than our `self.state.epoch`. If not, we can skip a clone, since `shared.state()` will clone out the state.

Perhaps we can implement this by making `shared.state()` return a RwLockGuard instead.